src/app/components/collection-details/collection-details.component.html

```diff
@@ -1,27 +1,115 @@
 <div class="w-full h-full flex flex-col justify-center items-start">
-  <h2 class="font-tilt-neon text-2xl py-4 text-pale-brown">
-    {{ selectedCollection?.replace("-", " ")?.toUpperCase() }}
-  </h2>
+  @if (selectedCollection) {
+    <h2 class="font-tilt-neon text-2xl py-4 text-pale-brown font-bold">
+      {{ selectedCollection.length < 4 ? selectedCollection.toUpperCase() : selectedCollection.substring(0, 1).toUpperCase() + selectedCollection.substring(1, selectedCollection.length) }}
+    </h2>
+  }
   <div
-    class="books-container w-full h-full overlow-x-hidden overflow-y-scroll flex flex-row flex-wrap gap-4 justify-center items-center content-start"
+    class="books-container w-full h-full overflow-x-hidden overflow-y-scroll flex flex-row flex-wrap gap-4 justify-start items-center content-start"
   >
-    @for (book of books; track book.title) {
-    <div
-      [routerLink]="['book']"
-      class="book-container w-[30%] h-[35%] flex flex-col"
-    >
-      <div
-        class="bg-transparent border border-white w-full h-full rounded-md sm:w-1/5"
-      ></div>
-      <p class="text-wrap font-tilt-neon text-pale-brown font-bold text-lg">
-        {{ book.title }}
-      </p>
-
-      <p class="text-wrap font-tilt-neon text-amaranth text-sm">
-        {{ book.author }}
-      </p>
-    </div>
 
+    @if (selectedCollection?.replace("-", " ") === "currently reading") {
+      @if ((currentBooks$ | async)?.bookshelf!.currentlyReading) {
+        @for (book of (currentBooks$ | async)!.bookshelf!.currentlyReading; track book.title) {
+          <div
+            [routerLink]="['book/' + book.isbn ]"
+            class="book-container w-[30%] h-[35%] flex flex-col"
+          >
+            <div
+              class="bg-transparent w-full h-full rounded-md sm:w-1/5"
+            >
+              <img alt="Book Cover Image" [src]=book.cover_url class="rounded-md object-cover">
+            </div>
+            <div class="basis-1/2 align-self-end flex flex-col justify-center">
+              <p class="text-wrap font-tilt-neon text-pale-brown font-bold text-lg">
+                {{ book.title.length > 12 ? book.title.slice(0, 10) + "..." : book.title }}
+              </p>
+
+              <p class="text-wrap font-tilt-neon text-amaranth text-sm">
+                {{ book.author }}
+              </p>
+            </div>
+          </div>
+
+        }
+      }
+
+    } @else if (selectedCollection === "read") {
```

Comment:
The reason why I am separating it currently, is because I am using async pipe to access the respective array for each category in my state and also allowing certain categories to be empty without breaking the UI, I will definitley look into another solution if time allows me to do so.